src/test/kotlin/uk/co/craigbass/pratura/unit/AddProductSpec.kt

```diff
@@ -9,23 +9,33 @@ import uk.co.craigbass.pratura.testdouble.InMemoryProductGateway
 import uk.co.craigbass.pratura.usecase.AddProduct
 
 class AddProductSpec : Spek({
-
   val productGateway = InMemoryProductGateway()
-
+  val useCase = memoized { AddProduct(productGateway) }
```

Comment:
In `Spek` test framework, a memoized value is read-only and cleared between each test "it" block